File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.py
import heapq
import sys
input = sys.stdin.readline

n,m = map(int,input().split())
if n==0:
    print(0)
    sys.exit()
books = list(map(int, input().split()))
temp = 0
res = 0
for i in range(n):
    if temp+books[i]>m:
        res +=1
        temp = books[i]
    else:
        temp += books[i]
if temp>0:
    res +=1
print(res)

File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m  = Integer.parseInt(st.nextToken());
        if(n==0){
            System.out.println(0);
            return;
        }
        st = new StringTokenizer(br.readLine());
        int[] books = new int[n];
        for(int i=0;i<n;i++){
            books[i] = Integer.parseInt(st.nextToken());
        }
        int temp = 0;
        int res = 0;
        for(int i=0;i<n;i++){
            if(temp+books[i]>m){
                res++;
                temp = books[i];
            }else{
                temp += books[i];
            }
        }
        if(temp>0){
            res++;
        }
        System.out.println(res);

        
            }
        }
    